doc/contribute/guidelines.rst

```diff
@@ -429,6 +432,9 @@ following exceptions:
   comment, ``//``, is not allowed.
 * Use ``/**  */`` for doxygen comments that need to appear in the documentation.
 
+When there are differences between the guidelines above and the rules enforced
+by the ``clang-format`` configuration, the ``clang-format`` configuration takes
+precedence.
```

Comment:
This is contrary to how we have enforced this in the past. The documented code style takes precedence over clang-format.

Reply:
Indeed, clang-format is a **tool** we use to enforce our coding guidelines. A tool should not take precedence over the very thing that it attempts to facilitate.

Reply:
When the documented code styles takes precedence, then this potentially means code authors have to hand edit files after running clang-format to conform to the coding style.  This also relies on the reviewers to catch the coding style problems.

I would prefer that we just had a policy to always run clang-format, but that is likely controversial.

Reply:
Ack - switched the language to make the documented guidelines have precedence.